src/wp-includes/blocks.php

```diff
@@ -933,27 +963,53 @@ function traverse_and_serialize_block( $block, $callback ) {
+ * arguments, the first one before serializing a block, and the second one after serializing.
+ * If either callback returns a string value, it will be prepended and appended to the serialized
+ * block markup, respectively.
+ *
+ * The callbacks will receive a reference to the current block as their first argument, so that they
+ * can also modify it, and the current block's parent block as second argument. Finally, the
+ * `$pre_callback` receives the previous block, whereas the `$post_callback` receives
+ * the next block as third argument.
  *
- * This should be used when there is a need to modify the saved blocks.
- * Prefer `serialize_blocks` when preparing blocks to be saved to post content.
+ * Serialized blocks are returned including comment delimiters, and with all attributes serialized.
+ *
+ * This function should be used when there is a need to modify the saved blocks, or to inject markup
+ * into the return value. Prefer `serialize_blocks` when preparing blocks to be saved to post content.
  *
  * @since 6.4.0
  *
- * @see serialize_blocks()
+ * @see serialize_block()
  *
- * @param array[]  $blocks   An array of representative arrays of parsed block objects. See serialize_block().
- * @param callable $callback Callback to run on each block in the tree before serialization.
- *                           It is called with the following arguments: $block, $parent_block, $block_index, $chunk_index.
- * @return string String of rendered HTML.
+ * @param array    $block         A representative array of a single parsed block object. See WP_Block_Parser_Block.
+ * @param callable $pre_callback  Callback to run on each block in the tree before it is traversed and serialized.
+ *                                It is called with the following arguments: &$block, $parent_block, $previous_block.
+ *                                Its string return value will be prepended to the serialized block markup.
+ * @param callable $post_callback Callback to run on each block in the tree after it is traversed and serialized.
+ *                                It is called with the following arguments: &$block, $parent_block, $next_block.
+ *                                Its string return value will be appended to the serialized block markup.
+ * @return string Serialized block markup.
  */
-function traverse_and_serialize_blocks( $blocks, $callback ) {
+function traverse_and_serialize_blocks( $blocks, $pre_callback = null, $post_callback = null ) {
 	$result = '';
 	foreach ( $blocks as $block ) {
-		// At the top level, there is no parent block, block index, or chunk index to pass to the callback.
-		$block = call_user_func( $callback, $block );
-		$result .= traverse_and_serialize_block( $block, $callback );
+		if ( is_callable( $pre_callback ) ) {
+			$result .= call_user_func_array(
+				$pre_callback,
+				array( &$block ) // At the top level, there is no parent block to pass to the callback.
```

Comment:
What about the previous or next for the top-level processing?

Reply:
Yeah, we might wanna add that 👍